# Boston House dataset

In this notebook, we are going to compare obtain results from the boston house dataset using our engine with the most upvoted notebook from Kaggle ( https://www.kaggle.com/vikrishnan/house-sales-price-using-regression ).

## I - Prediction from our engine 

In [1]:
from automl import prediction
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [2]:
df = pd.read_csv(
    'https://archive.ics.uci.edu/ml/machine-learning-databases/housing/housing.data', header=None, sep='\s+')
df.columns = ['CRIM', 'ZN', 'INDUS', 'CHAS', 'NOX', 'RM',
              'AGE', 'DIS', 'RAD', 'TAX', 'PTRATIO', 'B', 'LSTAT', 'MEDV']

In [3]:
a = prediction(df, 'MEDV')

NaNs represent 0.0 pourcentage of dataset
there is 52.96442687747036 pourcetage of rows with outliers
first model:
{'model': AdaBoostRegressor(base_estimator=None, learning_rate=1.0, loss='linear',
         n_estimators=50, random_state=None), 'accurracy': {'accurracy': 0.8112931987619452, 'rmse': 3.2959885412993692}, 'name': 'AdaBoostRegressor'}
second model:
{'model': RandomForestRegressor(bootstrap=False, criterion='mse', max_depth=None,
           max_features='sqrt', max_leaf_nodes=None,
           min_impurity_decrease=0.0, min_impurity_split=None,
           min_samples_leaf=1, min_samples_split=5,
           min_weight_fraction_leaf=0.0, n_estimators=10, n_jobs=None,
           oob_score=False, random_state=None, verbose=0, warm_start=False), 'accurracy': {'accurracy': 0.7865962754147812, 'rmse': 2.551138287741539}, 'name': 'RandomForestRegressor'}
third model:
{'model': LinearSVR(C=0.95, dual=True, epsilon=0.2, fit_intercept=True,
     intercept_scaling=1.0, loss='epsilon_inse

We finally obtain a:
- RMSE of 3.29 with an accuracy of 0.81 using an Adaboost Regressor and,
- RMSE of 2.55 with an accuracy of 0.78 using a RandomForest Regressor.

## II - Kaggle results

In the Kaggle notebook, the author used a GradientBoosting Regressor and obtained a MSE of 26.32 which mean that he found a RMSE of 5.13.

## III - Conclusion

Our model found a RMSE value close to the one found by the Kaggle author, which means that our engine is properly working.